engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -564,6 +583,22 @@ static void DestroyTextureImage(dmGraphics::TextureImage& texture_image)
     delete[] texture_image.m_Alternatives.m_Data;
 }
 
+static void CheckTextureResource(lua_State* L, int i, const char* field_name, const char** texture_path_out, dmGraphics::HTexture* texture_out)
+{
+    lua_getfield(L, i, field_name);
+    if (!lua_isstring(L, -1))
+    {
+        luaL_error(L, "Unable to get texture resource from field '%s', string expected", field_name);
```

Comment:
This goes against the intended mechanism of DM_LUA_STACK_CHECK.
Either you let the caller handle the error, or you don't use DM_LUA_STACK_CHECK for this.